### PR TITLE
[Review] Add http header token support + extend tests to run offline

### PR DESCRIPTION
**Pull request by ziahamza**

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -20,30 +36,41 @@ params.forEach(function(val, index, array) {
     case '--url':
       proxyUrl = params[index + 1]
       break
+    case '--port':
```
Comment (sirvar) [inline src/cli.js]: With more params for CLI, I think we should use a lib to parse

Comment (ziahamza) [reply src/cli.js]: Yea we should. That refactor can probably come in a dedicated PR. We can make a Gh issue about that perhaps.

File: src/lib/index.test.js [hunk 2]
```diff
@@ -2,7 +2,184 @@ const createHandler = require('./index')
...
+  req.on('end', () => {
+    res.end(
+      JSON.stringify({
+        body
+      })
+    )
+  })
+}
+
 describe('Backend proxy lib', () => {
```
[the remark this answers is not in the document]
Comment (ziahamza) [reply src/lib/index.test.js]: weird, it passes in my machine. what error do you get @sirvar ?

[the remark this answers is not in the document]
Comment (ziahamza) [reply src/lib/index.test.js]: weird. what error do you get?

[the remark this answers is not in the document]
Comment (sirvar) [reply src/lib/index.test.js]:
```
  ● Backend proxy lib › Test proxy › Proxy JSON properly

    getaddrinfo ENOTFOUND reqres.in reqres.in:443
      
      at errnoException (dns.js:50:10)
      at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:92:26)

  ● Backend proxy lib › Test proxy › Get body properly

    socket hang up
      
      at createHangUpError (_http_client.js:345:15)
      at TLSSocket.socketCloseListener (_http_client.js:377:23)
      at emitOne (events.js:120:20)
      at TLSSocket.emit (events.js:210:7)
      at _handle.close (net.js:549:12)
      at TCP.done [as _onclose] (_tls_wrap.js:350:7)
```

[the remark this answers is not in the document]
Comment (ziahamza) [reply src/lib/index.test.js]: @sirvar Can you post your NodeJS version, and try the request manually from your network?

Thats why I have added offline reproducable specs too 😅 ... so we can kinda remove these old specs as other tests cover up pretty much everything they do (and more)

**Comment by ziahamza** [comment]
@sirvar I further added CORS support (so proxy can be used in browser properly), and also removed the `request` dependency, so the project now has officially no other dependencies are should be pretty fast to install on NPM! 🎉

**Comment by ziahamza** [comment]
@arslanarshad31 @sirvar would love review for the changes, and push it to NPM.

**Pull request merged**